src/tools/clippy/clippy_lints/src/utils/internal_lints/unnecessary_def_path.rs

```diff
@@ -325,18 +325,9 @@ fn inherent_def_path_res(cx: &LateContext<'_>, segments: &[&str]) -> Option<DefI
     })
 }
 
-fn get_lang_item_name(cx: &LateContext<'_>, def_id: DefId) -> Option<Symbol> {
-    if let Some(lang_item) = cx.tcx.lang_items().items().iter().position(|id| *id == Some(def_id)) {
-        let lang_items = def_path_res(cx, &["rustc_hir", "lang_items", "LangItem"], Some(Namespace::TypeNS)).def_id();
-        let item_name = cx
-            .tcx
-            .adt_def(lang_items)
-            .variants()
-            .iter()
-            .nth(lang_item)
-            .unwrap()
-            .name;
-        Some(item_name)
+fn get_lang_item_name(cx: &LateContext<'_>, def_id: DefId) -> Option<&'static str> {
+    if let Some((lang_item, _)) = cx.tcx.lang_items().iter().find(|(_, id)| *id == def_id) {
+        Some(lang_item.variant_name())
```

Comment:
Are the `LangItem` variant names interned? Then I could return `sym::$variant` in `variant_name` and use the symbol.

Reply:
Ah I think now I got your question. The symbol name for e.g. `Sized` is `sized`. But we want to create a suggestion that tells the Clippy developer to use `LangItem::Sized` instead of `match_def_path(.., ["path", "to", "Sized"])`

Reply:
ah. You could propose `cx.tcx.lang_items().owned_box()` instead of `cx.tcx.lang_items().get(LangItem::OwnedBox)`, which always uses the symbol name for the method name.

Reply:
Oh that is an even better idea! Thanks, I will update this PR ASAP (which might be this weekend :| ).

Reply:
@oli-obk Looking at this again, we have utility functions in Clippy that take a `LangItem`.

```rust
pub fn is_res_lang_ctor(cx: &LateContext<'_>, res: Res, lang_item: LangItem) -> bool {
```

and

```rust
pub fn is_res_lang_ctor(cx: &LateContext<'_>, res: Res, lang_item: LangItem) -> bool {
```

For those we could no longer produce a applicable suggestion. I mean this lint is internal, so placeholder suggestion might be enough here. I will do that for now until I can come up with a better way to do this.

Reply:
these should be able to take the symbol name and compare against `Some(res.def_id())` if i'm understanding their purpose

Reply:
Yes, but I just noticed that the `symbol_name` is also not the method name, that is used to get the `LangItem` `DefId`. 

https://github.com/rust-lang/rust/blob/b6097f2e1b2ca62e188ba53cf43bd66b06b36915/compiler/rustc_hir/src/lang_items.rs#L283

There is also currently no way to get the method name.

Reply:
I could also refactor those functions to take a `DefId`. But that would make the API uglier. And just doing that to be able to produce a suggestion sounds wrong to me.

Reply:
So without this change, there is no way to produce a reasonable suggestion anymore. It is an internal lint, so I can just remove the suggestion, but I would like to avoid this.

Reply:
Oof ok, yea I see now. Let's go with this PR then, it's not a relevant maintenance cost. Just add a comment that this is all for clippy